packages/devtools-utils/src/worker-utils.js

```diff
@@ -57,17 +57,21 @@ WorkerDispatcher.prototype = {
       });
     };
   }
-}
+};
 
 function workerHandler(publicInterface: Object) {
-  return function workerHandler(msg: Message) {
+  return function (msg: Message) {
     const { id, method, args } = msg.data;
-    const response = publicInterface[method].apply(undefined, args);
-    if (response instanceof Promise) {
-      response.then(val => self.postMessage({ id, response: val }),
-                    err => self.postMessage({ id, error: err }));
-    } else {
-      self.postMessage({ id, response });
+    try {
+      const response = publicInterface[method].apply(undefined, args);
+      if (response instanceof Promise) {
+        response.then(val => self.postMessage({ id, response: val }),
+                      err => self.postMessage({ id, error: err }));
+      } else {
+        self.postMessage({ id, response });
+      }
+    } catch (error) {
+      self.postMessage({ id, error });
     }
```

Comment:
it might be nice to add unit tests where there are exceptions that are caught here.